### PR TITLE
Filter private, local only, or non routeable ip addresses

### PR DESCRIPTION
**Pull request by HM4704**
This PR
- filters private, local only, or non routeable ip addresses from peering
- fixes some unit tests

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -181,6 +181,12 @@ github.com/huin/goupnp v1.3.0/go.mod h1:gnGPsThkYa7bFi/KWmEysQRf48l2dvR5bxr2OFck
 github.com/inconshreveable/mousetrap v1.0.0/go.mod h1:PxqpIevigyE2G7u3NXJIT2ANytuPF1OarO4DADm73n8=
 github.com/inconshreveable/mousetrap v1.1.0 h1:wN+x4NVGpMsO7ErUn/mUI3vEoE6Jt13X2s0bqwp9tc8=
 github.com/inconshreveable/mousetrap v1.1.0/go.mod h1:vpF70FUmC8bwa3OWnCshd2FqLfsEA9PFc4w1p2J65bw=
+github.com/iotaledger/hive.go/constraints v0.0.0-20240517131232-748f1ce3a2d2 h1:5s73eiIv88KICanZK2mKSXXE/k0GrqE1tkint9nijds=
+github.com/iotaledger/hive.go/constraints v0.0.0-20240517131232-748f1ce3a2d2/go.mod h1:JF7jjkL6tSUOXm23SWadBzBrl7eJk1DQRLc/fNoVZ+o=
+github.com/iotaledger/hive.go/ierrors v0.0.0-20240517131232-748f1ce3a2d2 h1:Qci9UOR7NUkmtMGce8QFkdUDvPlQRWNjiEjzsO8bQtk=
+github.com/iotaledger/hive.go/ierrors v0.0.0-20240517131232-748f1ce3a2d2/go.mod h1:GQY0/35sjgT9Poi1Vrs9kFVvAkuKzGXfVh4j6CBXsAA=
+github.com/iotaledger/hive.go/lo v0.0.0-20240520064018-c635e5900894 h1:b3r0G8WQ1HBQeObTg2j7hbgJjb2oCgSVaIWazFZgVSQ=
+github.com/iotaledger/hive.go/lo v0.0.0-20240520064018-c635e5900894/go.mod h1:AJRiMLXVOLKvtuuCKBqe3zBeIB6qbp1vZrU1rPHm1CU=
```
Comment (lunfardo314) [inline go.sum:189]: pls remove any dependencies to iotaledger

File: peering/filter.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	for _, addr := range filtersToApply {
+		f, err := mamask.NewMask(addr)
+		if err != nil {
+			panic(fmt.Sprintf("unable to parse ip mask filter %s: %s", addr, err))
+		}
+		filters.AddFilter(*f, multiaddr.ActionDeny)
+	}
+
+	return func(addresses []multiaddr.Multiaddr) []multiaddr.Multiaddr {
+		return lo.Filter(addresses, func(m multiaddr.Multiaddr) bool {
```
Comment (lunfardo314) [inline peering/filter.go:67]: this is the only place which requires hive.go. Pls remove. Instead `util.PurgeSlice()` can be used. Proxima is minimalistic on dependencies

File: peering/filter.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+import (
+	"fmt"
+
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/multiformats/go-multiaddr"
+	mamask "github.com/whyrusleeping/multiaddr-filter"
+)
+
+// copied from https://github.com/iotaledger/iota-core
```
Comment (lunfardo314) [inline peering/filter.go:11]: do we need to mention it? It is taken from somewhere else anyway

File: peering/peers.go [hunk 4]
```diff
@@ -298,6 +302,11 @@ func readPeeringConfig() (*Config, error) {
 
 	cfg.IgnoreAllPullRequests = viper.GetBool("peering.ignore_pull_requests")
 	cfg.AcceptPullRequestsFromStaticPeersOnly = viper.GetBool("peering.pull_requests_from_static_peers_only")
+	if viper.IsSet("peering.allow_local_ips") {
```
Comment (lunfardo314) [inline peering/peers.go:305]: why not simply `cfg.AllowLocalIPs = viper.GetBool("peering.allow_local_ips")` ?

**Review by lunfardo314 — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (HM4704) [reply go.sum:189]: Done.

File: peering/filter.go (hunk 2 above)
Comment (HM4704) [reply peering/filter.go:67]: Done.

File: peering/peers.go (hunk 4 above)
Comment (HM4704) [reply peering/peers.go:305]: Done.

File: peering/filter.go (hunk 3 above)
Comment (HM4704) [reply peering/filter.go:11]: Removed.

**Review by lunfardo314 — APPROVED** [review]

**Pull request merged**